cmd/kubeadm/app/util/version.go

```diff
@@ -62,7 +65,13 @@ var (
 //  latest-1    (latest release in 1.x, including alpha/beta)
 //  latest-1.0  (and similarly 1.1, 1.2, 1.3, ...)
 func KubernetesReleaseVersion(version string) (string, error) {
-	return kubernetesReleaseVersion(version, fetchFromURL)
+	var err error
+	versionLock.Lock()
+	defer versionLock.Unlock()
+	if versionCache[version] == "" {
+		versionCache[version], err = kubernetesReleaseVersion(version, fetchFromURL)
+	}
+	return versionCache[version], err
```

Comment:
if i recall correctly, under normal user execution of a kubeadm command this KubernetesReleaseVersion function would be called once. so while this would fix the unit tests its not something we should add in the main code.

i think this function is called when a unit test creates a ClusterConfiguration with an empty "kubernetesVersion" field?
would including this version field skip the calling of this function?